parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java

```diff
@@ -219,6 +228,191 @@ public void testTwoLevelList() throws IOException {
     assertThat(recordRead.get("topbytes")).isEqualTo(expectedBinary);
   }
 
+  @Test
+  public void testStreamClosedProperly() throws IOException {
+    // test for input
+    {
+      class TestStreamClosedProperlyStream extends SeekableInputStream
+          implements DelegatingInputStream {
+        boolean thisClosed = false;
+        boolean delegateClosed = false;
+
+        {
+          reset();
+        }
+
+        @Override
+        public InputStream getDelegate() {
+          return new FSDataInputStream(
+              new InputStream() {
+                @Override
+                public int read() throws IOException {
+                  return 0;
+                }
+
+                @Override
+                public void close() throws IOException {
+                  delegateClosed = true;
+                }
+              });
+        }
+
+        @Override
+        public long getPos() throws IOException {
+          return 0;
+        }
+
+        @Override
+        public void seek(long newPos) throws IOException {}
+
+        @Override
+        public int read() throws IOException {
+          return 0;
+        }
+
+        @Override
+        public void close() throws IOException {
+          thisClosed = true;
+          delegateClosed = true;
+        }
+
+        public void reset() {
+          thisClosed = false;
+          delegateClosed = false;
+        }
+      }
+
+      try (TestStreamClosedProperlyStream stream = new TestStreamClosedProperlyStream()) {
+        try (org.apache.parquet.io.SeekableInputStream _unused =
```

Comment:
please separate this into their own test methods, where one test methods makes sure to cover the input stream and the other the output stream